feat(cloud-sql): Add connection sample for SQL Server/node-mssql

shubha-rajan opened the pull request:
Fixes #1589

fhinkel commented:
Thanks, can you add the kokoro config as well please

kurtisvg commented on cloud-sql/sqlserver/mssql/server.js:
```diff
@@ -0,0 +1,197 @@
+// Copyright 2019 Google LLC
```
```suggestion
// Copyright 2020 Google LLC
```

Don't forget to change this everywhere

shubha-rajan replied:
did a global find/replace to change all of them.

kurtisvg commented on cloud-sql/sqlserver/mssql/server.js:
```diff
@@ -0,0 +1,197 @@
+  config.user = process.env.DB_USER; // e.g. 'my-db-user'
+  config.password = process.env.DB_PASS; // e.g. 'my-db-password'
+  config.database = process.env.DB_NAME; // e.g. 'my-database'
+  config.server = process.env.DEPLOYED ? '172.17.0.1' : '127.0.0.1';
```
Can we add a comment to this clarifying App Engine Flex uses `172.17.0.1`?

shubha-rajan replied:
done

kurtisvg commented on cloud-sql/sqlserver/mssql/server.js:
```diff
@@ -0,0 +1,197 @@
+  // [END cloud_sql_server_mssql_limit]
+
+  // [START cloud_sql_server_mssql_lifetime]
+  // 'idleTimeoutMillis' is the number of milliseconds a connection must sit idle in the pool 
+  // and not be checked out before it is automatically closed
+  config.pool.idleTimeoutMillis = 600000,
```
Idle timeout for the other samples is under "[Connection timeout](https://cloud.google.com/sql/docs/mysql/manage-connections#pools)".

This is different than "Connection duration", which the MySQL sample doesn't have (may need to do something similar here): 
https://cloud.google.com/sql/docs/mysql/manage-connections#duration

shubha-rajan replied:
moved, and also removed the `lifetime` region tag

kurtisvg commented on cloud-sql/sqlserver/mssql/test/server.test.js:
```diff
@@ -0,0 +1,47 @@
+// Copyright 2018 Google LLC
```
```suggestion
// Copyright 2020 Google LLC
```

kurtisvg commented:
@shubha-rajan LGTM - let me know if you need me to take a look at the kokoro config.

shubha-rajan commented:
@fhinkel I added the kokoro config and updated `build.sh`, but I'm not sure how to create a new SQL server instance to test the sample against or where to add the instance connection name to the stored secrets (looks like all the secrets are read from a storage bucket?). Can you take a look please?

fhinkel commented:
Let's wait until the CL is merged so we see if the new CI test is run and passes. Thanks!

shubha-rajan commented:
The new test passes, and the 1 failing test is unrelated to this PR and not required, so I'm going to merge.

shubha-rajan commented:
Wait, after checking the logs, it looks like the test errors out, but still shows up as a success. I'll investigate this.
![image](https://user-images.githubusercontent.com/19631367/80841736-e7f21e80-8bb4-11ea-99a3-db2d2ec56123.png)

shubha-rajan commented:
Merging this now, will resolve test failures in #1783
